src/lsps2/service.rs

```diff
@@ -240,14 +241,16 @@ impl OutboundJITChannel {
 
 struct PeerState {
 	outbound_channels_by_scid: HashMap<u64, OutboundJITChannel>,
+	scid_by_user_channel_id: HashMap<u128, u64>,
 	pending_requests: HashMap<RequestId, LSPS2Request>,
 }
 
 impl PeerState {
 	fn new() -> Self {
 		let outbound_channels_by_scid = HashMap::new();
 		let pending_requests = HashMap::new();
-		Self { outbound_channels_by_scid, pending_requests }
+		let scid_by_user_channel_id = HashMap::new();
```

Comment:
We can't because with ChannelReady we only get the user channel id to lookup the OutboundJITChannel. 

Before this change we were making the user set user channel id to the scid but that's also strange and changed in this PR.